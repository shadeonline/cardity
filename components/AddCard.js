import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { Text, View, Image, TouchableOpacity } from 'react-native';
import { Cell } from 'react-native-tableview-simple';

const AddCard = () => {
  const navigation = useNavigation();
  // Check if redirect is blank ornot if redirect is not blank then redirect it to the desired view, else do nth
  const handlePress = () => {
    navigation.navigate('Plans');
  };

  return (
    <Cell
      contentContainerStyle={styles.containerStyle}
      cellContentView={
        <TouchableOpacity onPress={handlePress} style={styles.cardContainer}>
          <View style={styles.cardContent}>
            <Text style={styles.plusSign}>+</Text>
          </View>
        </TouchableOpacity>
      }
    />
  );
};

const styles = {

  containerStyle: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 290, 
    width: '100%',
    backgroundColor: 'transparent',
    highlightColor: '#ccc',
    paddingTop: 25,
    paddingBottom: 25,
  },
  cellContentView: {
    backgroundColor: 'transparent',
  },
  cardContainer: {
    backgroundColor: 'white',
    borderRadius: 10,
    borderWidth: 2,
    borderColor: 'purple',
    padding: 20,
    width: '100%',
    // Add any other styling properties you need
  },
  cardContent: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    // Add any other styling properties you need
  },
  plusSign: {
    fontSize: 24,
    color: 'purple',
  },
  cardTitleContainer: {
    position: 'absolute',
    top: '8%',
    left: '8%',
    backgroundColor: 'rgba(255, 255, 255, 0.9)',
    alignItems: 'center',
  },
};

export default AddCard;
