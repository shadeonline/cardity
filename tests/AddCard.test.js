import React from 'react';
import { render } from 'react-native-testing-library';
import { NavigationContainer } from '@react-navigation/native';
import AddCard from '../components/AddCard';


describe('CreateTaskButton', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <AddCard />
      </NavigationContainer>
    );

    // Use Jest's expect to match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});