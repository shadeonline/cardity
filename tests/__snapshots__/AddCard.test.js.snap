// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTaskButton matches snapshot 1`] = `
<View>
  <RCTSafeAreaView
    style={
      [
        {
          "backgroundColor": "#FFF",
        },
        {
          "flexGrow": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "minHeight": 44,
            "paddingLeft": 15,
            "paddingRight": 20,
          },
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 290,
            "highlightColor": "#ccc",
            "justifyContent": "center",
            "paddingBottom": 25,
            "paddingTop": 25,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "white",
            "borderColor": "purple",
            "borderRadius": 10,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 20,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "purple",
                "fontSize": 24,
              }
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
